runtime/src/main/java/org/corfudb/runtime/collections/ISMRMap.java

```diff
@@ -272,4 +273,55 @@ default void undoPutAll(ISMRMap<K,V> map, Map<K,V> undoRecord,
     @Override
     Set<Entry<K, V>> entrySet();
 
+    @DontInstrument
+    Object getMapGarbageManager();
+
+    /**
+     * This class is used to maintain the location and other information on different keys.
+     * @param <T> the type of the map key.
+     */
+    class SMRMapGarbageManager<T>{
```

Comment:
I think this class should not be put in this interface (ISMRMap).  CorfuTable does not implement this interface.

Reply:
I have moved this class to a dedicated file.